src/lib/features/metrics/instance/metrics.ts

```diff
@@ -143,12 +143,26 @@ export default class ClientMetricsController extends Controller {
             try {
                 const promises: Promise<void>[] = [];
                 for (const app of applications) {
-                    promises.push(
-                        this.clientInstanceService.registerClient(
-                            app,
-                            clientIp,
-                        ),
-                    );
+                    if (
+                        app.sdkType === 'frontend' &&
+                        typeof app.sdkVersion === 'string' &&
+                        this.flagResolver.isEnabled('registerFrontendClient')
+                    ) {
+                        this.clientInstanceService.registerFrontendClient({
```

Comment:
frontend registration is more lightweight since it doesn't send any events to eventBus/eventStore